src/models/Call.ts

```diff
@@ -215,25 +216,38 @@ export abstract class Call extends TypedEventEmitter<CallEvent, CallEventHandler
      */
     public async start(_params?: WidgetGenerationParameters): Promise<void> {
         const messagingStore = WidgetMessagingStore.instance;
-        this.messaging = messagingStore.getMessagingForUid(this.widgetUid) ?? null;
-        if (!this.messaging) {
-            // The widget might still be initializing, so wait for it.
-            try {
-                await waitForEvent(
-                    messagingStore,
-                    WidgetMessagingStoreEvent.StoreMessaging,
-                    (uid: string, widgetApi: ClientWidgetApi) => {
-                        if (uid === this.widgetUid) {
-                            this.messaging = widgetApi;
-                            return true;
-                        }
-                        return false;
-                    },
+        let messaging = messagingStore.getMessagingForUid(this.widgetUid);
+        const startTime = performance.now();
+        // The widget might still be initializing, so wait for it in an async
+        // event loop. We need the messaging to be both present and started, so
+        // we register listeners for both cases. Note that due to React strict
+        // mode, the messaging could even be aborted and replaced by an entirely
+        // new messaging while we are waiting here!
+        while (!messaging?.widgetApi) {
```

Comment:
I am a bit lost, if messaging is `undefined` this is then a `while(true)` what is breaking the loop?

Reply:
...nothing, I think this needs a different check.